### PR TITLE
Fix String-to-bytes and bytes-to-String for UTF-8 Pt. 2

### LINKED ISSUE #4765: Fix String-to-bytes and bytes-toString for UTF-8 for all remaining code
As a follow-on to #3815, which fixed these conversions for all non-test code in the 2.1 branch for 2.1.3, the same should be done for all code in the main/3.1 branch, and should include test code as well (at the very least to set a good example for developers coming along later; we should just have a habit of being rigorous with encoding issues regardless of its location).

### PR DESCRIPTION
**Pull request by meatballspaghetti**
This PR attempts to address all String-to-bytes and bytes-to-String conversions so they all consistently use UTF-8 (see more commentary in #3815). Being a follow-on to #3815, this PR mostly makes changes to test code that had not previously been covered.

The following fixes have been made:
- String-to-bytes: `String.getBytes()` -> `String.getBytes(UTF_8)`
- Bytes-to-String: `new String(byte[])` -> `new String(byte[], UTF_8)`

Fixes #4765 "Fix String-to-bytes and bytes-toString for UTF-8 for all remaining code"

### CODE REVIEW

**Review by DomGarguilo — APPROVED** [review]

File: core/src/test/java/org/apache/accumulo/core/data/KeyTest.java [hunk 1]
```diff
@@ -39,10 +40,14 @@ public class KeyTest {
 
   @Test
   public void testDeletedCompare() {
-    Key k1 = new Key("r1".getBytes(), "cf".getBytes(), "cq".getBytes(), new byte[0], 0, false);
-    Key k2 = new Key("r1".getBytes(), "cf".getBytes(), "cq".getBytes(), new byte[0], 0, false);
-    Key k3 = new Key("r1".getBytes(), "cf".getBytes(), "cq".getBytes(), new byte[0], 0, true);
-    Key k4 = new Key("r1".getBytes(), "cf".getBytes(), "cq".getBytes(), new byte[0], 0, true);
+    Key k1 = new Key("r1".getBytes(UTF_8), "cf".getBytes(UTF_8), "cq".getBytes(UTF_8), new byte[0],
```
[the remark this answers is not in the document]
Comment (meatballspaghetti) [reply core/src/test/java/org/apache/accumulo/core/data/KeyTest.java:43]: There are assertEquals statements a couple lines below to confirm that each pair of identical Key objects are indeed identical (they are not used outside of this particular unit test). If I reduce these four variables to two, those assertEquals statements would no longer apply. Should I proceed with that and delete the assertions, or leave all as is?

[the remark this answers is not in the document]
Comment (DomGarguilo) [reply core/src/test/java/org/apache/accumulo/core/data/KeyTest.java:43]: Sorry I was pretty vague in my comment. I meant making new variables for the row, cf and cq that are used to construct the four `Key`s. For example:
```java
final byte[] row = "r1".getBytes(UTF_8);
final byte[] cf = "cf".getBytes(UTF_8);
final byte[] cq = "cq".getBytes(UTF_8);
Key k2 = new Key(row, cf, cq, new byte[0], 0, false);
```
Either way I think the code is fine as is and no changes need to made unless you want to.

[the remark this answers is not in the document]
Comment (meatballspaghetti) [reply core/src/test/java/org/apache/accumulo/core/data/KeyTest.java:43]: I see what you're saying, I agree with it. I can implement/push the change once I can connect to an instance again.

**Comment by meatballspaghetti** [comment]
> Changes look good and consistent with #3815.
> 
> One thing that may have been missed is changing instances of the method ref equivalents: `String::new` and `String::getBytes` use UTF_8. I found instances of `String::new` that could be changed, but no instances of `String::getBytes`.

Maybe I'm misunderstanding, but `String::new` just creates an empty String with no encoding involved and no conversion from bytes. Are you suggesting that I convert each instance of `String::new` to `s -> s.getBytes(UTF_8)`? Since they're just empty Strings to being with, I wasn't sure if that would be applicable here.

File: core/src/test/java/org/apache/accumulo/core/data/KeyTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (meatballspaghetti) [reply core/src/test/java/org/apache/accumulo/core/data/KeyTest.java]: Done, updated.

**Comment by kevinrr888** [comment]
@meatballspaghetti - Here is an example of what I'm talking about from `TokenFileIT`:
```
int i = scanner.stream().map(Map.Entry::getValue).map(Value::get).map(String::new)
            .map(Integer::parseInt).collect(onlyElement());
```
Here, `String::new` is used as a method reference to the `String` constructor that takes `byte[]` as its arg since `Value::get` returns `byte[]`. I'm suggesting the above (and elsewhere this applies) be changed to
```
int i = scanner.stream().map(Map.Entry::getValue).map(Value::get).map(e -> new String(e, UTF_8))
                .map(Integer::parseInt).collect(onlyElement());
```

**Comment by meatballspaghetti** [comment]
@kevinrr888 Makes sense, thanks. Updated.

**Review by ctubbsii — APPROVED** [review]
Looks good to me, but since this is for 2.1, I'll wait to merge it until I can be sure I'm prepared to merge it into the main branch and elasticity and work through any merge conflicts that might arise.

**Review by kevinrr888 — APPROVED** [review]
LGTM

**Comment by ctubbsii** [comment]
> > Looks good to me, but since this is for 2.1, I'll wait to merge it until I can be sure I'm prepared to merge it into the main branch and elasticity and work through any merge conflicts that might arise.
> 
> Sorry I did not see this before merging into 2.1. Just saw the approvals and went ahead and merged it. I will attempt to merge it up into main and elasticity. Were there any specific issues or difficulties you foresaw with the merge or did you just mean you didn't have the time to do it when the comment was made so you were planning on doing it later?

I just didn't have time to do it when I reviewed it, and was planning on doing it later, but if it's already done, then that's not a problem at all!

**Pull request merged**